substrate/client/network/sync/src/lib.rs

```diff
@@ -3364,4 +3364,129 @@ mod test {
 		pending_responses.remove(&peers[1]);
 		assert_eq!(pending_responses.len(), 0);
 	}
+
+	/// The test demonstrates the issue discussed here:
+	/// https://github.com/paritytech/polkadot-sdk/issues/493#issuecomment-1776648354.
```

Comment:
Could you create an issue for this and reference it here instead of the comment? Copy-pasting from the comment would be enough, may be adding some other observations you have already made.